[Select] Update doc for select component

thyhmdo opened the pull request:
Closes #3458 and #3459 

Updated a small amount of content and major issues on images in Usage and Style tabs for Select.

laurenmrice reviewed (changes requested):
Looks good Thy! Just a couple comments:

----

### Usage tab

**Select versus Dropdown section:**
- I wonder for this image if the label for the component should say `Select` on the left and `Dropdown` on the right to make it more clear which component is which. Or instead of changing the image, in the caption below it clarify that Select is on the left and Dropdown is on the right.
- For the Dropdown on the right, we do not currently have the prompt in the dropdown field also shown as an option in the dropdown list. Need to clarify with Mike G on this, I think it was another option of how we could structure the Select component, not the Dropdown component.
- For the Dropdown list in this image, the bottom rule under the hover state should not be visible.
<img width="1063" alt="Screen Shot 2023-05-04 at 9 28 32 AM" src="https://user-images.githubusercontent.com/43969356/236219704-5a944485-3f31-4eba-b37b-ff88fc3a4924.png">

**Validation: Default selection section:**
- I think we should reword this paragraph to be something like below to be more clear on what Carbon offers as a default but what users could modify on their end based on their specific needs:

> The select field has an empty selected option by default, and the user can decide whether to choose a different option from the list. Depending on the use case, this defaulted behavior can be modified to be a prefilled selection to the first option in the list, which is typically shown in alphabetical order, or the prefilled selection could be a frequent or commonly used option that is in the list.

**States section image:**
- Just a small thing, but you can take out the word state in the labels and just have the state name to keep it more consistent and concise with how we are treating our other component state images. This will also apply to the images on the Style tab.
- I think the Disabled state (default style) image should be empty since the rest are empty and it would match the fluid style image.
- The Disabled state label (fluid style) image should say `Disabled`, not read-only.
- The Read-only state (fluid style) should have `$text-secondary` input text.
<img width="862" alt="Screen Shot 2023-05-04 at 9 46 36 AM" src="https://user-images.githubusercontent.com/43969356/236225331-819ba9b1-24ef-41e5-9b35-a768024d43a1.png">

----

### Style tab

**Interactive states image:**
- The Disabled state (default style) should not have a border-bottom.
- The Read-only state (fluid style) should have `$text-secondary` input text.

laurenmrice commented on src/pages/components/select/usage.mdx:
```diff
@@ -290,49 +290,48 @@ appropriately for its context.
 - Helper text is optional but replaced with warning or error text when these
   states appear.
 
 #### Order
```
```suggestion
#### List order
```
This may be more clear that we are referencing the list option order.

thyhmdo commented:
@laurenmrice Thanks for the feedback! 

I updated most of the feedback you suggested. There are few things: 
- I liked how you rephrased the part for the paragraphed for modifying the defaulted option. I just took that out from the Validation section since it's down at the bottom and moved it to the Anatomy part instead
- The dropdown example is confirmed by Mike Gower, the only beed he had was the Dropdown selected option is not highlighted enough. He'd prefer better options or ref. [this issue here](https://github.com/carbon-design-system/carbon/discussions/7929#discussioncomment-456319). Though it's not in the code yet so I'm hesitant to update that in the image here. 
- I'd keep the Validation image as it is right now. I think it's not in the State section so it doesn't have to have the state label. It's also about showing an example of validation so maybe it makes sense as it as.

andreancardona reviewed (approved):
looks good! 🎉

alisonjoseph reviewed (approved):
LGTM!

laurenmrice commented on src/pages/components/select/usage.mdx:
```diff
@@ -211,17 +211,21 @@ Inline selects have less visual weight on a page because they are borderless.
    options.
-2. **Default option:** Usually a frequent option that users would choose from a
-   select list.
+2. **Default option:** An empty option selected by default. The user can decide
```
Should the # 2  "Default option" label be called "Selected option" instead? In the image example above, it is just a little confusing because there is already something selected from the list, and it isn't an empty selection which is the default behavior.

laurenmrice commented on src/pages/components/select/usage.mdx:
```diff
@@ -290,49 +294,48 @@ appropriately for its context.
 The select has multiple interactive states for both default and inline variants:
-**enabled**, **hover**, **focus**, **open**, **error**, **warning**,
-**disabled**, and **read-only**.
+**enabled**, **hover**, **selected**, **focus**, **open**, **error**,
```
There could be a row in the table to represent and describe the `Open state` since it's mentioned in the paragraph above and in the image.

laurenmrice commented on src/pages/components/select/usage.mdx:
```diff
@@ -345,7 +348,7 @@ The select has multiple interactive states for both default and inline variants:
 <Tab label="Default">
 
-![Select interactive states](images/select-states-fixed.png)
+![Select interactive states](images/select-states-default.png)
```
Read-only state (fixed): You can take out the helper text here since you don’t show it for the other states.

laurenmrice commented on src/pages/components/select/style.mdx:
```diff
@@ -82,13 +82,13 @@ tabs: ['Usage', 'Style', 'Code', 'Accessibility']
 <Tab label="Default">
 
-![Select interactive states](images/select-style-states-fixed.png)
+![Interactive states of default select](images/select-style-interactive-states-default.png)
```
Disabled and Read-only state (fixed): You can take out the helper text for these since it is not in the other states.

laurenmrice commented on src/pages/components/select/style.mdx:
```diff
@@ -169,7 +169,7 @@ and any proper nouns capitalized. Select text should be three words or less.
 <div className="image--fixed">
 
-![Structure and spacing measurements for Inline Select](images/select-style-structure-default-fluid.png)
+![Structure and spacing measurements for fluid select](images/select-style-structure-fluid.png)
```
This image is missing the teal padding overlay on the right side for the error state.
<img width="1006" alt="Screen Shot 2023-05-10 at 3 24 21 PM" src="https://github.com/carbon-design-system/carbon-website/assets/43969356/40a554be-bd7a-4671-8f93-9f14232b8c97">

laurenmrice reviewed (changes requested):
Just a couple more things!

----

**Overview > Select versus Dropdown section**

- I would mention in this paragraph that you are referring to the `list` styling of the component.
> Another important difference between the two components is the underlying code. The select component’s list appearance will be determined by the browser being used, while the dropdown component’s list appearance can be styled as needed.

- For the dropdown example in this image, use our current dropdown list without the repeated prompt in the list as an option. This isn't implemented yet in code, and we should figure out if that is the direction we want to take with dropdown across the board.

<img width="1071" alt="Screen Shot 2023-05-10 at 3 10 41 PM" src="https://github.com/carbon-design-system/carbon-website/assets/43969356/86d9b97c-ddb4-448f-a66d-d5de9474175d">
